src/openrct2/ride/Vehicle.cpp

```diff
@@ -584,6 +584,20 @@ void Vehicle::MoveRelativeDistance(int32_t distance)
     ClearFlag(VehicleFlags::MoveSingleCar | VehicleFlags::CollisionDisabled);
 }
 
+void Vehicle::UpdateTrackChange()
```

Comment:
This is all only ever useful for your one scripting bit of code it would make more sense to keep it with the rest of the scripting function.

Reply:
`_vehicleCurPosition` namely.

Reply:
> `_vehicleCurPosition` namely.

Pretty sure that is a global var

Reply:
well but it's still only accessible from Vehicle.cpp if it's defined there. But nevermind that because I just tested and it works well with just using a local variable instead. dunno why it's being done like that in the existing functions.

But... on my way trying to move it to `ScVehicle` I ran into couple of `Vehicle` functions that really are private.
`GetTrackProgress/GetMoveInfo/GetRideTypeDescriptor`. What do you suggest?

Reply:
I think the reason all the other users are using a global Is just because thats how it worked in assembly. I'd need to have a look see how all this is getting updated. Perhaps there is a different function we can make that can be used by a few other consumers.  I'm just not a fan of having basically half a function in one file and the other half in another.